cluster-service/Makefile

```diff
@@ -29,28 +24,82 @@ deploy:
+# be redesigned when we move to use Helm templates.
+# *Warning* if you change any parameter remember to change it in both the deploy and the deploy-using-azure-db targets
+# if they apply to both
+deploy-using-azure-db: configure-tmp-provision-shard deploy-namespace-template deploy-istio-configurations-template deploy-secrets-template-using-azure-db
 	AZURE_CS_MI_CLIENT_ID=$(shell az identity show \
 			-g ${RESOURCEGROUP} \
 			-n clusters-service \
 			--query clientId) && \
-	oc process --local -f deploy/integration/cluster-service-namespace.yaml \
-	-p CLIENT_ID=$${AZURE_CS_MI_CLIENT_ID} | oc apply -f -
+	AZURE_FIRST_PARTY_APPLICATION_CLIENT_ID=${AZURE_FIRST_PARTY_APPLICATION_CLIENT_ID} && \
+	CS_SERVICE_PRINCIPAL_CREDS_BASE64='$(shell az keyvault secret show --vault-name "service-kv-aro-hcp-dev" --name "aro-hcp-dev-sp-cs" | jq .value -r | base64 | tr -d '\n')' && \
+	TENANT_ID=$(shell az account show --query tenantId --output tsv) && \
+	OIDC_BLOB_SERVICE_ENDPOINT=$(shell az storage account list --query "[?starts_with(name, 'arohcpoidc')].primaryEndpoints.blob" -g ${REGIONAL_RESOURCEGROUP} -o tsv) && \
+	OIDC_WEB_ENDPOINT=$(shell az storage account list --query "[?starts_with(name, 'arohcpoidc')].primaryEndpoints.web" -g ${REGIONAL_RESOURCEGROUP} -o tsv) && \
+	oc process --local -f deploy/openshift-templates/arohcp-service-template.yml \
+		-p AZURE_CS_MI_CLIENT_ID=$${AZURE_CS_MI_CLIENT_ID} \
+		-p TENANT_ID=$${TENANT_ID} \
+		-p REGION=${REGION} \
+		-p SERVICE_KEYVAULT_NAME=${KEYVAULT_NAME} \
+		-p CS_SERVICE_PRINCIPAL_CREDS_BASE64=$${CS_SERVICE_PRINCIPAL_CREDS_BASE64} \
+		-p IMAGE_REGISTRY=${ARO_HCP_IMAGE_ACR}.azurecr.io \
+		-p IMAGE_REPOSITORY=app-sre/uhc-clusters-service \
+		-p AZURE_FIRST_PARTY_APPLICATION_CLIENT_ID=$${AZURE_FIRST_PARTY_APPLICATION_CLIENT_ID} \
+		-p FPA_CERT_NAME=${FPA_CERT_NAME} \
+		-p IMAGE_TAG=${CLUSTERS_SERVICE_IMAGE_TAG} \
+		-p DATABASE_DISABLE_TLS=false \
+		-p DATABASE_AUTH_METHOD="az-entra" | oc apply -f -
+
+deploy-integ:
+	AZURE_CS_MI_CLIENT_ID=$(shell az identity show \
+		-g ${RESOURCEGROUP} \
+		-n clusters-service \
+		--query clientId) && \
+		oc process --local -f deploy/integration/cluster-service-namespace.yaml \
+			-p CLIENT_ID=$${AZURE_CS_MI_CLIENT_ID} | oc apply -f -
+
+deploy-namespace-template:
+	ISTO_VERSION=$(shell az aks list --query "[?tags.clusterType == 'svc-cluster' && starts_with(resourceGroup, '${RESOURCEGROUP}')].serviceMeshProfile.istio.revisions[-1]" -o tsv)  && \
+	oc process --local -f deploy/openshift-templates/arohcp-namespace-template.yml \
+		-p ISTIO_VERSION=$${ISTO_VERSION} | oc apply -f -
+
+deploy-istio-configurations-template:
+	kubectl apply -f deploy/istio.yml
+
+deploy-secrets-template: configure-tmp-provision-shard
+	oc process --local -f deploy/openshift-templates/arohcp-secrets-template.yml \
+		-p PROVISION_SHARDS_CONFIG="$$( base64 -i deploy/tmp-provisioning-shard.yml)" | oc apply -f -
```

Comment:
Does this need to configure a db password? Otherwise the password will be `Password for the PostgreSQL connection user` 🤔  Alternatively, rather than having this as a default value, it can just be an empty string? And this make target then can set the db password?

Reply:
thanks, saw the update, lgtm